### PR TITLE
Read/write to Firestore database

### LINKED ISSUE #1: 1. As a component, I want to be able to read from and write to the Firestore database so that users can persist information.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* The following have been added as project dependencies: `firebase` & `react-firestore`
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

### PR DESCRIPTION
**Pull request by alexandra-lim**
## Description

- Removed Create React App boilerplate
- Installed Firebase and Firestore dependencies
- Created a counter that increments by 1 with each button click
- Connected Firestore to store click count in database
- Update database count and have new count reflected on front end 

## Related Issue

Closes #1 

## Acceptance Criteria

* The following have been added as project dependencies: `firebase` & `react-firestore`
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

image

### After

Screen Shot 2020-10-07 at 1 16 19 PM

Screen Shot 2020-10-07 at 1 16 35 PM

## Testing Steps / QA Criteria

1.  From your terminal, pull down this branch (`git pull origin eb-al-connect-to-firestore`) and checkout to the branch (`git checkout eb-al-connect-to-firestore`)
2. Then install the newly added dependencies by running `npm install`
3.  Open up [Firestore Database](https://console.firebase.google.com/u/0/project/tcl-12-smart-shopping-list/firestore/data~2Fcount~2FFQyeXUHAfge9AXp6qXrK)
4. Click on the counter button and check to see if the count updates on the page and in the database.
5. Edit the count directly in the Firestore database by editing the number value and refresh the app. The count on the app should now reflect the value in the database.

### CODE REVIEW

File: src/Components/Counter.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  useEffect(() => {
+    newCount.get().then(function (doc) {
+      setCount(doc.data().number);
+    });
+  }, [newCount]);
+
+  const increaseCount = (newCount) => {
+    setCount(count + 1);
+    newCount.update({ number: count + 1 });
+  };
```
Comment (terchiem) [inline src/Components/Counter.js]: There might be a case where the update to the Firestore might not go through and the counts will be out of sync. You can try updating the Firestore first and then use `then` to update the browser count if it is successful and `catch` to log any errors if it isn't successful.

**Comment by GabbyJ** [comment]
This looks really nice. Good job!
We are noticing a flicker in the count occasionally on Chrome and Safari, but not Firefox. We tried testing to see if the browser was updating before the database, but that doesn't seem to be causing it. Below it happens on 30-31, 36-37, and 47-48.
count-flicker

**Comment by alexandra-lim** [comment]
> This looks really nice. Good job!
> We are noticing a flicker in the count occasionally on Chrome and Safari, but not Firefox. We tried testing to see if the browser was updating before the database, but that doesn't seem to be causing it. Below it happens on 30-31, 36-37, and 47-48.
> count-flicker

Good catch @GabbyJ and @terchiem! We're not sure what could be causing this issue. We'd be curious to hear from you or the mentors on how we can address this issue.

File: src/Components/Counter.js (hunk 1 above)
Comment (edenbekele) [reply src/Components/Counter.js]: Thanks for the feedback! We added `then` and `catch` to both our get request and our update request to catch any errors that we may have!

File: src/Components/Counter.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      .catch((error) => {
+        console.log('Error getting document:', error);
+      });
+  }, [newCount]);
+
+  const increaseCount = (newCount) => {
+    newCount
+      .update({ number: count + 1 })
+      .then(() => {
+        setCount(count + 1);
```
Comment (mxmason) [inline src/Components/Counter.js:23]: Just to capture what we discussed in our workshop: there's a way to reliably track your data without reusing `+ 1` as we do here. Don't let this be a blocker right now – you know you can update and read the state, and that's the main goal! We can always work on elegant solutions later.

File: src/Components/Counter.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  const newCount = db.collection('count').doc('FQyeXUHAfge9AXp6qXrK');
+
+  useEffect(() => {
+    newCount
+      .get()
+      .then(function (doc) {
+        setCount(doc.data().number);
+      })
+      .catch((error) => {
+        console.log('Error getting document:', error);
```
Comment (mxmason) [inline src/Components/Counter.js:15]: Glad to see you're handling errors! ✨

**Review by mxmason — APPROVED** [review]
Great work, team! I left a couple comments; nothing blocking. Keep it up!

**Pull request merged**